lib/Serialization/ModuleFile.cpp

```diff
@@ -230,11 +218,13 @@ Status ModuleFile::associateWithFileContext(FileUnit *file, SourceLoc diagLoc,
           modulePath.front().Item == file->getParentModule()->getName()) {
         return error(Status::MissingUnderlyingModule);
       }
-
       // Otherwise, continue trying to load dependencies, so that we can list
       // everything that's missing.
```

Comment:
I think this makes more sense after the new `if` now?

Reply:
That comment is about the previous block and continues the comment before it `// If we're missing the module we're an overlay for, treat that specially.`. I moved the empty line around to make that more obvious, I think it helps a bit in the final result.